src/Microsoft.FeatureManagement/ServiceCollectionExtensions.cs

```diff
@@ -64,5 +88,75 @@ public static IFeatureManagementBuilder AddFeatureManagement(this IServiceCollec
 
             return services.AddFeatureManagement();
         }
+
+        /// <summary>
+        /// Adds scoped <see cref="FeatureManager"/> and other required feature management services.
+        /// </summary>
+        /// <param name="services">The service collection that feature management services are added to.</param>
+        /// <returns>A <see cref="IFeatureManagementBuilder"/> that can be used to customize feature management functionality.</returns>
+        /// <exception cref="FeatureManagementException">Thrown if <see cref="FeatureManager"/> has been registered as singleton.</exception>
+        public static IFeatureManagementBuilder AddScopedFeatureManagement(this IServiceCollection services)
+        {
+            if (services.Any(descriptor => descriptor.ServiceType == typeof(FeatureManager) && descriptor.Lifetime == ServiceLifetime.Singleton))
+            {
+                throw new FeatureManagementException(
+                    FeatureManagementError.Conflict,
+                    "Singleton feature management has been registered.");
+            }
+
+            services.AddLogging();
+
+            services.AddMemoryCache();
+
+            //
+            // Add required services
+            services.TryAddSingleton<IFeatureDefinitionProvider, ConfigurationFeatureDefinitionProvider>();
+
+            services.AddScoped(sp => new FeatureManager(
```

Comment:
I don't see a reason to add `FeatureManager` into DI. Previously we only registered IFeatureManager.

Reply:
In preview branch, we add FeatureManager and use it for IFeatureManager and IVariantFeatureManager.
https://github.com/microsoft/FeatureManagement-Dotnet/blob/preview/src/Microsoft.FeatureManagement/ServiceCollectionExtensions.cs#L34
I follow that way.